### DISCUSSION TITLE
Including some hidden files and directories in the search

### DISCUSSION THREAD

**Discussion by nordlow** [BurntSushi/ripgrep#2011] [Q&A] [answered]
Generally, how do prevent ripgrep from _skipping_ a specific set of (hidden) files and directories (starting with a dot), in my case `.gitlab-ci.yml`?

Alternatively, how do I make ripgrep ignore only a certain of files and directories such VCS directories and file patterns registered in .gitignore.

I'm not certain which filtering technique that is the most compact to express.

I tried

```sh
rg --hidden --iglob '.git' ...
```

because I never (seldom) want to search .git directories  but that also ignores the .gitlab-ci.yml file. I've scanned through the docs but couldn't find a flag that suited my needs. I'm missing a flag `--iglob-regex` I could use as

```sh
rg --hidden --iglob-regex '^.git$' ...
```

or alternatively

```sh
rg --glob-regex '^.gitlab-ci.yml$' ...
```

And

```sh
rg --glob '.gitlab-ci.yml' ...
```

is not usable to me as it skips other files.

I simply want the default behavior of `rg` except I want to specify a set of files starting with a dot to be included in the search. In the general case I believe the least surprising behavior (correct?) for `rg` would be to search hidden files _if_ they are registered by .git (added and commited) as in my case.

**Comment by BurntSushi** [comment, marked as answer]
Can you just add `!.gitlab-ci.yml` to a `.ignore` file? That should whitelist it and include it in your searches for the directory in which the `.ignore` file resides.

**Reply by nordlow** [reply]
Thanks. That works but it's not ideal because I need to pollute quite a lot of repos with this file. I still strongly believe the vc-registered files should be searched by default regardless of whether they start with a dot or not or whether or not it's marked as hidden in the filesystem.

**Reply by BurntSushi** [reply]
ripgrep doesn't work by asking which files are tracked. It works by looking at ignore files. (This has been reported before. See: https://github.com/BurntSushi/ripgrep/issues/340#issuecomment-274786392.)

You don't need to "pollute a bunch of repos" with a `.ignore` file for this. You can put one global `.ignore` file somewhere and then add it to your ripgrep config file with `--ignore-file`.

Even if ripgrep knew which files your VC system tracked, the `--hidden` flag would still do what it says it does: it would ignore hidden files.
